### PR TITLE
Fix Whisper Positional Embeddings when using decoder context

### PR DESCRIPTION
**Pull request by andyehrenberg**

### CODE REVIEW

**Comment by sgugger** [comment]
cc @ArthurZucker

File: src/transformers/generation/utils.py [hunk 1]
```diff
@@ -878,6 +878,7 @@ def _get_logits_processor(
                 SuppressTokensAtBeginLogitsProcessor(generation_config.begin_suppress_tokens, begin_index)
             )
         if generation_config.forced_decoder_ids is not None:
+            forced_decoder_ids = [[input_ids_seq_length + i[0] - 1, i[1]] for i in forced_decoder_ids]
             processors.append(ForceTokensLogitsProcessor(generation_config.forced_decoder_ids))
```
Comment (ArthurZucker) [inline src/transformers/generation/utils.py:882]: this should either be used or not included? 
 ```suggestion
            processors.append(ForceTokensLogitsProcessor(generation_config.forced_decoder_ids))
```

**Review by ArthurZucker — APPROVED** [review]
LGTM thanks for fixing this. I just have a question as to why you need to modify the logit processing?

**Comment by barbieri** [comment]
is this getting in or really not needed?

**Comment by ArthurZucker** [comment]
Seems to work well without 😉

**Pull request closed without merging**